multiversx_sdk_cli/delegation/staking_provider.py

```diff
@@ -1,169 +1,347 @@
+        tx.options = int(args.options)
+        tx.guardian = args.guardian
+        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
+
+        if args.gas_limit:
+            tx.gas_limit = int(args.gas_limit)
+
+        return tx
+
+    def get_transaction_for_setting_metadata(self, owner: IAccount, args: Any) -> ITransaction:
+        delegation_contract = Address.new_from_bech32(args.delegation_contract)
+
+        tx = self._factory.create_transaction_for_setting_metadata(
+            sender=owner.address,
+            delegation_contract=delegation_contract,
+            name=args.name,
+            website=args.website,
+            identifier=args.identifier
+        )
+        tx.nonce = int(args.nonce)
+        tx.version = int(args.version)
+        tx.options = int(args.options)
+        tx.guardian = args.guardian
+        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
+
+        if args.gas_limit:
+            tx.gas_limit = int(args.gas_limit)
+
+        return tx
+
+    def _parse_public_bls_keys(self, public_bls_keys: str) -> List[ValidatorPublicKey]:
+        keys = public_bls_keys.split(",")
+        validator_public_keys: List[ValidatorPublicKey] = []
+
+        for key in keys:
+            validator_public_keys.append(ValidatorPublicKey(bytes.fromhex(key)))
+
+        return validator_public_keys
+
+    def _get_public_keys_and_signed_messages(self, args: Any) -> Tuple[List[ValidatorPublicKey], List[bytes]]:
+        validators_file_path = Path(args.validators_file).expanduser()
+        validators_file = ValidatorsFile(validators_file_path)
+
+        pubkey = self._get_pubkey_for_signing(args)
+
+        public_keys: List[ValidatorPublicKey] = []
+        signed_messages: List[bytes] = []
+        for validator in validators_file.get_validators_list():
+            # Get path of "pemFile", make it absolute
+            validator_pem = Path(validator.get("pemFile")).expanduser()
+            validator_pem = validator_pem if validator_pem.is_absolute() else validators_file_path.parent / validator_pem
+
+            pem_file = ValidatorPEM.from_file(validator_pem)
+
+            validator_signer = ValidatorSigner(pem_file.secret_key)
+            signed_message = validator_signer.sign(pubkey)
+
+            public_keys.append(pem_file.secret_key.generate_public_key())
+            signed_messages.append(signed_message)
+
+        return public_keys, signed_messages
+
+    def _get_pubkey_for_signing(self, args: Any) -> bytes:
+        account = Account()
+        if args.using_delegation_manager:
+            account = Account(address=Address.new_from_bech32(args.delegation_contract))
+        elif args.pem:
+            account = Account(pem_file=args.pem)
+        elif args.keyfile:
+            password = load_password(args)
+            account = Account(key_file=args.keyfile, password=password)
+
+        return account.address.get_public_key()
+
+    def __get_empty_transaction(self) -> Transaction:
```

Comment:
Instead, can we do `else raise Error("...")`?

Reply:
That sounds good. Done!